fgpyo/sam/__init__.py

```diff
@@ -899,3 +900,14 @@ def __next__(self) -> Template:
         name = self._iter.peek().query_name
         recs = self._iter.takewhile(lambda r: r.query_name == name)
         return Template.build(recs, validate=False)
+
+
+class SamOrder(enum.Enum):
```

Comment:
Having looked at this many months later, I wonder if this is too simplistic (see our Scala equivalent: https://github.com/fulcrumgenomics/fgbio/blob/ff1ca67772c34a8defeb835d692a53ad6e536f74/src/main/scala/com/fulcrumgenomics/bam/api/SamOrder.scala#L172).  It doesn't allow custom sort orders, including setting group order, sub-sort, etc.  I am fine with this for now, but changing this later will require breaking changes, so lets require @tfenne to weigh in too.

Reply:
I would stop where we are - there's nothing here that would prevent us from doing something more sophisticated later.